pkg/gofr/datasource/file/ftp/fs.go

```diff
@@ -0,0 +1,217 @@
+package ftp
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"gofr.dev/pkg/gofr/datasource"
+
+	"github.com/jlaffaye/ftp"
+)
+
+// ftpFileSystem represents a file system interface over FTP.
+type ftpFileSystem struct {
+	file   *ftpFile          // Pointer to a single file object (not used in the provided methods)
+	conn   ServerConn        // FTP server connection
+	config *Config           // FTP configuration
+	logger datasource.Logger // Logger interface for logging
+}
+
+// Config represents the FTP configuration.
+type Config struct {
+	Host      string // FTP server hostname
+	User      string // FTP username
+	Password  string // FTP password
+	Port      string // FTP port
+	RemoteDir string // Remote directory path
```

Comment:
This will also be the base path for all the operation, should be mentioned in the comments